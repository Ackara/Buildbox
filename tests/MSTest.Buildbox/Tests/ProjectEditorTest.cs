using Acklann.Buildbox.Versioning;
using Acklann.Buildbox.Versioning.Editors;
using ApprovalTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.IO;
using System.Linq;

namespace MSTest.Buildbox.Tests
{
    [TestClass]
    public class ProjectEditorTest
    {
        [TestMethod]
        public void Update_should_edit_a_netstardard_project_file_metadata()
        {
            // Arrange
            var manifest = CreateManifest();
            var sut = new NetStandardProjectEditor();

            // Act
            string tempPath = Path.GetTempFileName();
            var projectFile = sut.FindProjectFile(TestDataDirectory.FullPath).First();
            projectFile.CopyTo(tempPath, overwrite: true);
            sut.Update(manifest, new FileInfo(tempPath));

            // Assert
            Approvals.VerifyXml(File.ReadAllText(tempPath));
        }

        [TestMethod]
        public void Update_should_edit_a_dotnet_project_file_metadata()
        {
            // Arrange
            var manifest = CreateManifest();
            var sut = new DotNetProjectEditor();

            // Act
            string tempPath = Path.Combine(Path.GetTempPath(), $"{nameof(Update_should_edit_a_dotnet_project_file_metadata)}.txt");
            var projectFile = sut.FindProjectFile(TestDataDirectory.FullPath).First();
            projectFile.CopyTo(tempPath, overwrite: true);
            sut.Update(manifest, new FileInfo(tempPath));

            // Assert
            Approvals.VerifyFile(tempPath);
        }

        [TestMethod]
        public void Update_should_edit_a_vsix_manifest_file()
        {
            // Arrange
            var manifest = CreateManifest();
            var sut = new VsixManifestEditor();

            // Act
            string tempPath = Path.Combine(Path.GetTempPath(), $"{nameof(Update_should_edit_a_vsix_manifest_file)}.xml");
            var projectFile = sut.FindProjectFile(TestDataDirectory.FullPath).First();
            projectFile.CopyTo(tempPath, overwrite: true);
            sut.Update(manifest, new FileInfo(tempPath));

            // Assert
            Approvals.VerifyFile(tempPath);
        }

        private static Manifest CreateManifest()
        {
            var manifest = new Manifest()
            {
                Title = "Buildbox",
                Version = new VersionInfo(1, 2, 3),
                Authors = "Seto Kiaba",
                Owner = "Kiaba Corp",
                Copyright = "All rights reserved by Me",
                Description = "This is for a automated test",
                ProjectUrl = "https://gihub.com/Ackara/Buildbox",
                IconUri = "https://example.com/icon/buildbox.png",
                LicenseUri = "https://example.com/buildbox/license.txt",
                Tags = "powershell, deploy, ci"
            };

            return manifest;
        }
    }
}