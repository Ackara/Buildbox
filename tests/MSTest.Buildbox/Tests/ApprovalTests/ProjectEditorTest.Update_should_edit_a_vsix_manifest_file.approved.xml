<?xml version="1.0" encoding="utf-8"?>
<PackageManifest Version="2.0.0" xmlns="http://schemas.microsoft.com/developer/vsx-schema/2011" xmlns:d="http://schemas.microsoft.com/developer/vsx-schema-design/2011">
  <Metadata>
    <Identity Id="Optimus.VSIX.411f03e3-dac2-4bcb-91ea-7e2cc488bfa6" Version="1.2.3" Language="en-US" Publisher="Kiaba Corp" />
    <DisplayName>Buildbox</DisplayName>
    <Description>This is for a automated test</Description>
  <License>https://example.com/buildbox/license.txt</License><Icon>https://example.com/icon/buildbox.png</Icon><Tags>powershell, deploy, ci</Tags></Metadata>
  <Installation>
    <InstallationTarget Id="Microsoft.VisualStudio.Community" Version="[15.0]" />
  </Installation>
  <Dependencies>
    <Dependency Id="Microsoft.Framework.NDP" DisplayName="Microsoft .NET Framework" d:Source="Manual" Version="[4.5,)" />
  </Dependencies>
  <Prerequisites>
    <Prerequisite Id="Microsoft.VisualStudio.Component.CoreEditor" Version="[15.0,16.0)" DisplayName="Visual Studio core editor" />
  </Prerequisites>
</PackageManifest>